test/y2network/hostname_writer_test.rb

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require_relative "../test_helper"
+require "y2network/sysconfig/hostname_writer"
+require "y2network/hostname"
+
+describe Y2Network::Sysconfig::HostnameWriter do
+  subject { Y2Network::Sysconfig::HostnameWriter.new }
+
+  describe ".update_hostname" do
```

Comment:
Update hostname is private, and you are really testing the write call that is ok.

```suggestion
  describe "#write" do
```